lib/precompiler.js

```diff
@@ -0,0 +1,111 @@
+var path = require('path');
+var resolveFrom = require('resolve-from');
+var resolve = require('resolve');
+var md5hex = require('md5-hex');
+var detective = require('babel-plugin-detective');
+var fs = require('graceful-fs');
+
+var base = path.join(__dirname, '..', 'index.js');
+
+var _babelCore;
+
+function babelCore() {
+	if (!_babelCore) {
+		_babelCore = require('babel-core');
+	}
+	return _babelCore;
+}
+
+function Precompiler(cacheDir) {
+	this._cacheDir = cacheDir;
+	this._cache = {};
+}
+
+Precompiler.prototype.cachePath = function (filename) {
+	return path.join(this._cacheDir, filename);
+};
+
+Precompiler.prototype.buildEntry = function (filename) {
+	if (this._cache[filename]) {
+		return;
+	}
+
+	var entry = this._cache[filename] = {};
+
+	var result = babelCore().transformFileSync(filename, {
+		plugins: [detective]
```

Comment:
> this will implicitly be merged with the .babelrc.

Yes. That's the idea. That is only precompiling your sources. We already precompile your tests in `caching-precompiler`.